WordPress/Classes/ViewRelated/Blog/SiteSettingsViewController.m

```diff
@@ -686,102 +678,50 @@ - (void)togglePushNotifications:(id)sender
     NSRegularExpression *wpadmin = [NSRegularExpression regularExpressionWithPattern:@"/wp-admin/?$" options:NSRegularExpressionCaseInsensitive error:&error];
     NSRegularExpression *trailingslash = [NSRegularExpression regularExpressionWithPattern:@"/?$" options:NSRegularExpressionCaseInsensitive error:&error];
-
+    
     urlToValidate = [wplogin stringByReplacingMatchesInString:urlToValidate options:0 range:NSMakeRange(0, [urlToValidate length]) withTemplate:@""];
     urlToValidate = [wpadmin stringByReplacingMatchesInString:urlToValidate options:0 range:NSMakeRange(0, [urlToValidate length]) withTemplate:@""];
     urlToValidate = [trailingslash stringByReplacingMatchesInString:urlToValidate options:0 range:NSMakeRange(0, [urlToValidate length]) withTemplate:@""];
-
+    
     return urlToValidate;
 }
 
-- (void)validateXmlprcURL:(NSURL *)xmlRpcURL
+- (void)validateLoginCredentials
 {
+    [SVProgressHUD showWithStatus:NSLocalizedString(@"Authenticating", @"") maskType:SVProgressHUDMaskTypeBlack];
+
+    NSURL *xmlRpcURL = [NSURL URLWithString:self.blog.xmlrpc];
     WordPressXMLRPCApi *api = [WordPressXMLRPCApi apiWithXMLRPCEndpoint:xmlRpcURL
                                                                username:self.username
                                                                password:self.password];
-
+    __weak __typeof__(self) weakSelf = self;
     [api getBlogOptionsWithSuccess:^(id options){
-        if ([options objectForKey:@"wordpress.com"] != nil) {
-            self.isSiteDotCom = YES;
-            [self loginForSiteWithXmlRpcUrl:[NSURL URLWithString:@"https://wordpress.com/xmlrpc.php"]];
-        } else {
-            self.isSiteDotCom = NO;
-            [self loginForSiteWithXmlRpcUrl:xmlRpcURL];
-        }
-    } failure:^(NSError *failure){
-        [SVProgressHUD dismiss];
-        [self validationDidFail:failure];
-    }];
-}
-
-- (void)loginForSiteWithXmlRpcUrl:(NSURL *)xmlRpcURL
-{
-    WordPressXMLRPCApi *api = [WordPressXMLRPCApi apiWithXMLRPCEndpoint:xmlRpcURL username:self.usernameTextCell.detailTextLabel.text password:self.password];
-    [api getBlogsWithSuccess:^(NSArray *blogs) {
-        [SVProgressHUD dismiss];
-        [self validationSuccess:[xmlRpcURL absoluteString]];
-    } failure:^(NSError *error) {
-        [SVProgressHUD dismiss];
-        [self validationDidFail:error];
-    }];
-}
-
-- (void)checkURL
-{
-    NSString *urlToValidate = [self getURLToValidate];
-
-    DDLogInfo(@"%@ %@ %@", self, NSStringFromSelector(_cmd), urlToValidate);
-
-    [SVProgressHUD showWithStatus:NSLocalizedString(@"Authenticating", @"") maskType:SVProgressHUDMaskTypeBlack];
-    [WordPressXMLRPCApi guessXMLRPCURLForSite:urlToValidate success:^(NSURL *xmlrpcURL) {
-        [self validateXmlprcURL:xmlrpcURL];
+        __typeof__(self) strongSelf = weakSelf;
+        strongSelf.blog.password = self.password;
+        [strongSelf.blog.managedObjectContext save:nil];
```

Comment:
@koke I think you are right all this need to move inside the BlogService. But I've looked at this a bit more, and I think that this change is not enough to update the password. This will update the current password in the keychain, but I also need to reset the api object stored inside the blog in order to get a new WordPressXMLRPCApi with the new password.